configs/components/runtime.rb

```diff
@@ -37,7 +37,17 @@
 
     # Curl is dynamically linking against zlib, so we need to include this file until we
     # update curl to statically link against zlib
-    pkg.install_file "#{settings[:tools_root]}/bin/zlib1.dll", "#{settings[:bindir]}/zlib1.dll"
+    pkg.install_file "#{settings[:tools_root]}/bin/zlib1.dll", "#{settings[:ruby_bindir]}/zlib1.dll"
+
+    # gdbm, yaml-cpp and iconv are all runtime dependancies of ruby, and their libraries need
+    # To exist inside our vendored ruby
+    # SPECIAL NOTE::: libffi is uneccessary in our vendored ruby here, because we install ffi
```

Comment:
^ Be careful about how the above is worded.  Our FFI gem does include it's own DLL (which as we discovered is an older version of `libFFI` than what Ruby bundles), but that doesn't mean that the Ruby vendored one is unnecessary.  It is used by Ruby's [`Fiddle module`](http://ruby-doc.org/stdlib-2.0.0/libdoc/fiddle/rdoc/Fiddle.html), which we've always shipped.

In other words, while we don't use `Fiddle` in our code, 3rd parties / end users could be using it in custom type and provider code, etc.  The chances are slim that that's the case I believe, but if we choose to later remove `libffi-6.dll` from our custom version, we should also remove `Fiddle` from Ruby, and it would need to be on a breaking version boundary.  I'm not sure it's worth the effort to do that.